pkg/app/config.go

```diff
@@ -295,6 +295,18 @@ func GenPackerVars(config Config, extraVarsPath string) ([]byte, error) {
 			case []string:
 				p[key] = strings.Join(v, ",")
 			default:
+				reflectOfV := reflect.ValueOf(v)
+				if reflectOfV.Kind() == reflect.Map {
+					m := make(map[string]interface{})
```

Comment:
why further conversion to `map[string]interface{}` needed? `p[key] = v` instead works as expected. Do we need to explicitly check if keys of the `run_tags` are string?

I tested with following
```go
  
default:
	reflectOfV := reflect.ValueOf(v)
	switch reflectOfV.Kind() {
	case reflect.Map:
		p[key] = v
	default:
		p[key] = fmt.Sprintf("%v", v)                           
```

Reply:
i think it would be make most sense to import a map of stings to values. it seems a bit more safe to be explicit when working with untyped objects

Reply:
I also found it hard to understand the need for reflect.

I guess the input (`v`) can only have a depth of two (keys, and subkeys), and when we copy it over, we want to skip any subkeys that aren't strings?

I guess this is because something like Shalin's suggestion, or this

```go
			case map[string]interface{}:
				p[key] = v
			default:
				p[key] = fmt.Sprintf("%v", v)
```

would panic if the subkey is a not a string, e.g., it is an integer.